src/plugins/discover/public/utils/get_fields_to_show.ts

```diff
@@ -9,23 +9,30 @@ import { getFieldSubtypeMulti } from '@kbn/data-views-plugin/public';
 import type { DataView } from '@kbn/data-views-plugin/public';
 
 export const getFieldsToShow = (fields: string[], dataView: DataView, showMultiFields: boolean) => {
+  if (showMultiFields) {
+    return fields;
+  }
+  const fieldMap = {} as Record<string, boolean>;
   const childParentFieldsMap = {} as Record<string, string>;
+  const parentFieldMap = {} as Record<string, boolean>;
```

Comment:
Why not make it just a set? The value is never read

Reply:
very true, I just applied to the pattern of this file, so I didn't want to introduce new stuff

Reply:
and it was a quick fix, that I just deployed to see if I break something

Reply:
it fixed the issue locally, that was enough info for me :)

Reply:
it might furthermore be possible to optimized, we have 2 big loops here, 1 forEach, 1 filter look. It might be possible to reduce to a single one, which would optimize a bit more

Reply:
@majagrubic I've refactored to use a set